pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -612,10 +600,6 @@ func (r *ReconcileJenkinsBaseConfiguration) compareContainers(expected corev1.Co
 		r.logger.Info(fmt.Sprintf("Resources have changed to '%+v' in container '%s', recreating pod", expected.Resources, expected.Name))
 		return true
 	}
-	if !reflect.DeepEqual(expected.SecurityContext, actual.SecurityContext) {
```

Comment:
This is not good idea. The main reason of the problem is that OpenShift modifies some part of the Jenkins pod. To fix it you have to check the value set by the OpenShift `kubectl get po jenkins-example -o yaml` and look at `spec.securityContext` and configure the same value in CR `spec.master.securityContext`.